data-pipeline/src/pipeline/comparator_sets.py

```diff
@@ -349,13 +298,79 @@ def select_top_set_urns(
     return urns
 
 
+def _map_pupil_comparator_set(row: pd.Series) -> bool:
+    """
+    Whether to generate a pupil comparator set for the row in question.
+
+    For partial-year data, financial and pupil data must be present.
+
+    For full-year data, pupil comparator sets are always generated.
+
+    :param row: grouped data
+    :return: whether to generate pupil comparator set
+    """
+    if not row["Partial Years Present"]:
+        return True
+
```

Comment:
I think this (and the buildings variant below) will need a couple of slight changes:

- if those columns aren't present, `return True` (as the academy data won't have these yet)
- if those columsn _are_ present, we need to check for `not False` rather than rely on truth-y (i.e. the all-schools/mixed data will have the columns by virtue of a `join()` but they'll be null)